extern/sector-storage/manager.go

```diff
@@ -187,6 +196,14 @@ func (m *Manager) AddLocalStorage(ctx context.Context, path string) error {
 	return nil
 }
 
+// AddWorker registers a worker with the miner's work scheduler.
+// We query the worker for it's current sessionID (a uuid generated by the worker on each startup)
+// to determine if we already know this current run of the worker and therefore discard it as a duplicate or if it's a new run and we need to add it
+// to our list of known workers.
+// We then repeatedly poll the worker's sessionID to detect restarts by identifying sessionID mismatches.
+// If we detect a worker restart, we will remove the current worker run from the list of known workers.
+// Upon restart, a worker will register itself again by calling `AddWorker` after generating a new session ID.
+// This polling basically acts as a heartbeat in the system.
```

Comment:
I would suggest to separate implementation details from the godoc that we want exposed publicly. This reads too imperative for a godoc. It feels like you are describing logic, and not function. The logic may change any time, so it's better described _inside_ as inline comments in the method body. What we want to describe in a godoc is function, expectations, prerequisites, and results.

Consider this style:

```
AddWorker registers a worker with the miner's work scheduler.

This method safely deduplicates workers we already know about. Thus,
a given worker will never be able to doubly register. Workers are
identified by their runtime session ID, which is randomly generated
upon restart.

Once added, the Manager will monitor a Worker through a polling
heartbeat. If the worker is gone, or there's a session ID mismatch,
the manager drops the worker. If the worker was restarted, it will
register again under the new session ID.
```